packages/common/src/utils/fileUtils.ts

```diff
@@ -0,0 +1,81 @@
+import fs from 'fs';
+import * as lodash from 'lodash-es';
+import { BadRequest } from './errors.js';
+
+const fsPromises = fs.promises;
+
+export type TsvRecordAsJsonObj = { [header: string]: string | string[] };
+
+export const ARRAY_DELIMITER_CHAR = '|';
+
+/**
+ * Validates if a file contains a .tsv extension
+ * @param file A file to check its extension
+ * throws a BadRequest errot type when file is not a .tsv
+ */
+export const validateTsvExtension = (file: Express.Multer.File): void => {
+	// check if has .tsv extension to prevent irregular file names from reaching service level
+	if (!file.originalname.match(/.*\.tsv$/)) {
+		throw new BadRequest('Invalid extension');
+	}
+};
+
+/**
+ * Reads a .tsv file and parse it to a JSON format
+ * @param {string} fileName
+ * @returns a JSON format objet
+ */
+export const tsvToJson = async (fileName: string): Promise<ReadonlyArray<TsvRecordAsJsonObj>> => {
+	const contents = await fsPromises.readFile(fileName, 'utf-8');
+	const arr = parseTsvToJson(contents);
+	return arr;
+};
+
+const parseTsvToJson = (content: string): ReadonlyArray<TsvRecordAsJsonObj> => {
```

Comment:
After some failed attempts to find a most robust library to parse TSVs to JSON, found more convenient to reuse most of this logic from argo-clinical.